### ISSUE TITLE
Baseline pCR Prediction via Radiomics (PyRadiomics)

### ISSUE THREAD

**Issue by annawoodard** [dsi-clinic/vanguard#2]
## What is the task?

Using only imaging data (and possibly your tumor segmentation masks), extract radiomic features from the tumor region with PyRadiomics, then train a simple classifier (e.g. logistic regression, random forest) to predict pCR (yes / no).

The model should be trained on a training split, evaluated on a held-out test split.

Report the AUC (area under ROC), and verify whether the model performs better than random (AUC > 0.5).

Produce a reproducible script or notebook that takes as input:
Paths to images and masks, and labels
and outputs predictions, metrics, and optionally calibration plots.

## Definition of Done

Code to take path to a train/test specification, load the data, train the model, validate on the test set, output a summary of the validation results.

### Feature Extraction
 - Use pyradiomics (radiomics feature extractor) to compute radiomic features for each patient’s tumor mask from one or more DCE phases (or aggregated image).
 - You may choose a default YAML parameter file or dictionary of featureClass, imageType, filter settings etc (per PyRadiomics usage).
 - Save the extracted features as a table (CSV or Parquet) with rows = patients and columns = features + patient ID + possibly other metadata.

### Train & Evaluate
- Train a baseline classifier (e.g. logistic regression with regularization, random forest) on the radiomics table (training split only).
- Evaluate on held-out test split and compute metrics: ROC AUC, confusion matrix, sensitivity/specificity at some threshold.
- Optionally provide calibration plot or PR curve.

### Result Logging
Write a metrics.json or metrics.yaml containing at least:

auc_test: <float>
auc_train: <float>
n_features_used: <int>
classifier_type: <string>

### Validation

Ensure AUC_test > 0.5 (i.e. better than chance). If the test AUC is ≤ 0.5, the run should still be allowed (for debugging), but the student must include a short commentary on failure (why it might fail, overfitting, feature quality, etc).

The reviewer should be able to run it with minimal instructions:

```
python baseline_pcr_radiomics.py --images /path/to/images --masks /path/to/masks \
  --labels labels.csv --split split.csv --output outdir
```

and see outputs appear (features file, metrics, predictions).

### Documentation / README
 - Add required dependencies to `requirements.txt`
 - Explain how to run the script (arguments, expected folder layout)
 - Document any parameter choices (e.g. binWidth, resampling, filter settings)
 - State limitations / caveats (small sample size, potential overfitting, need for normalization)
